File: tv/scripts/generate-audio.js
#!/usr/bin/env node

/**
 * Generate audio files for TV Ranking sound effects
 * Uses tone.js to create synthesized audio files
 */

const Tone = require('tone');
const fs = require('fs');
const path = require('path');

const outputDir = path.join(__dirname, '../public/audio');

// Ensure output directory exists
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

/**
 * Create a simple tone and save as audio
 * Since Tone.js can't directly save to MP3 without external tools,
 * we'll create a simple workaround using raw audio data
 */
async function generateAudioFiles() {
  console.log('Gerando arquivos de áudio...');

  // levelup.mp3 - Ascending tone (subida de posição)
  const levelupTone = new Tone.PolySynth(Tone.Synth, {
    oscillator: { type: 'triangle' },
    envelope: { attack: 0.005, decay: 0.1, sustain: 0, release: 0.1 }
  }).toDestination();

  // goal.mp3 - Success tone (meta atingida)
  const goalTone = new Tone.PolySynth(Tone.Synth, {
    oscillator: { type: 'sine' },
    envelope: { attack: 0.01, decay: 0.2, sustain: 0.1, release: 0.2 }
  }).toDestination();

  // champion.mp3 - Victory fanfare (novo campeão)
  const championTone = new Tone.PolySynth(Tone.Synth, {
    oscillator: { type: 'square' },
    envelope: { attack: 0.01, decay: 0.15, sustain: 0, release: 0.1 }
  }).toDestination();

  // overtake.mp3 - Quick alert (ultrapassagem)
  const overtakeTone = new Tone.PolySynth(Tone.Synth, {
    oscillator: { type: 'sawtooth' },
    envelope: { attack: 0.002, decay: 0.08, sustain: 0, release: 0.05 }
  }).toDestination();

  // alert.mp3 - Simple beep (alerta)
  const alertTone = new Tone.PolySynth(Tone.Synth, {
    oscillator: { type: 'sine' },
    envelope: { attack: 0.005, decay: 0.1, sustain: 0, release: 0.05 }
  }).toDestination();

  console.log('✓ Áudios gerados com sucesso');
  console.log(`📁 Salvos em: ${outputDir}`);
}

// Note: Direct MP3 export isn't available in Tone.js without additional tools
// This script demonstrates the structure for audio generation
// In production, audio files should be created with proper audio tools

console.log(`
⚠️  Nota: Para arquivos MP3 reais, use:
  - Tone.js com ffmpeg
  - Audacity (exportar como MP3)
  - SoundBible.com (sons libres)
  - Freepik.com (efeitos sonoros)

Arquivos de placeholder foram criados em:
${outputDir}

Você pode substituir manualmente com arquivos MP3 reais.
`);

// Create placeholder files so the app doesn't error
const sounds = ['levelup', 'goal', 'champion', 'overtake', 'alert', 'ambient'];
sounds.forEach(sound => {
  const filePath = path.join(outputDir, `${sound}.mp3`);
  if (!fs.existsSync(filePath)) {
    // Create empty placeholder files - in production these would be real audio
    fs.writeFileSync(filePath, Buffer.from([
      0xFF, 0xFB, 0x10, 0x00, // MP3 header (simple valid header)
      ...Array(100).fill(0) // Minimal MP3 frame data
    ]));
    console.log(`✓ ${sound}.mp3 (placeholder)`);
  }
});

console.log('\n✓ Script concluído!');

File: tv/scripts/generate_audio.py
#!/usr/bin/env python3
"""
Generate simple audio files for TV Ranking sound effects
Creates WAV files with synthesized tones
"""

import wave
import struct
import math
import os

def create_audio_file(filename, frequency, duration, output_dir='../public/audio'):
    """Create a simple sine wave audio file"""
    sample_rate = 44100
    num_samples = int(sample_rate * duration)

    # Ensure output directory exists
    os.makedirs(output_dir, exist_ok=True)
    filepath = os.path.join(output_dir, filename)

    with wave.open(filepath, 'w') as wav_file:
        wav_file.setnchannels(1)  # Mono
        wav_file.setsampwidth(2)  # 16-bit
        wav_file.setframerate(sample_rate)

        # Generate sine wave
        for i in range(num_samples):
            # Envelope (fade in/out)
            if i < sample_rate * 0.05:  # 50ms attack
                envelope = i / (sample_rate * 0.05)
            elif i > num_samples - sample_rate * 0.1:  # 100ms release
                envelope = (num_samples - i) / (sample_rate * 0.1)
            else:
                envelope = 1.0

            # Generate sine wave
            value = 32767 * 0.3 * envelope * math.sin(2 * math.pi * frequency * i / sample_rate)
            wav_file.writeframes(struct.pack('<h', int(value)))

    print(f'[OK] {filename}')

# Create audio files with different frequencies and durations
print('Gerando arquivos de áudio...\n')

sounds = {
    'levelup.wav': (440, 0.3),      # A4, ascending tone
    'goal.wav': (523, 0.5),         # C5, success tone
    'champion.wav': (659, 0.7),     # E5, victory tone
    'overtake.wav': (392, 0.2),     # G4, quick alert
    'alert.wav': (330, 0.15),       # E4, simple beep
    'ambient.wav': (261, 2.0),      # C4, background drone (longer)
}

os.chdir(os.path.dirname(os.path.abspath(__file__)))

for filename, (frequency, duration) in sounds.items():
    try:
        create_audio_file(filename, frequency, duration)
    except Exception as e:
        print(f'[ERROR] Erro ao criar {filename}: {e}')

print('\n[OK] Arquivos de áudio criados com sucesso!')
print('[DIR] Salvos em: public/audio/')
print('\n[INFO] Nota: Estes sao audios simples de teste.')
print('Para producao, substitua com audios reais de qualidade.')
